airflow/models/taskinstance.py

```diff
@@ -2034,20 +2034,22 @@ def generate_command(
     def log_url(self) -> str:
         """Log URL for TaskInstance."""
         run_id = quote(self.run_id)
-        base_date = quote(self.logical_date.strftime("%Y-%m-%dT%H:%M:%S%z"))
         base_url = conf.get_mandatory_value("webserver", "BASE_URL")
         map_index = f"&map_index={self.map_index}" if self.map_index >= 0 else ""
-        return (
+        _log_uri = (
             f"{base_url}"
             f"/dags"
             f"/{self.dag_id}"
             f"/grid"
             f"?dag_run_id={run_id}"
             f"&task_id={self.task_id}"
             f"{map_index}"
-            f"&base_date={base_date}"
             "&tab=logs"
         )
+        if self.logical_date:
+            base_date = quote(self.logical_date.strftime("%Y-%m-%dT%H:%M:%S%z"))
+            _log_uri = f"{_log_uri}&base_date={base_date}"
+        return _log_uri
```

Comment:
Let’s create an issue to track this. (I’m not sure what `base_date` even does.)

Reply:
this only prevents airflow from breaking but not actually solving the log not found issue

Reply:
https://github.com/apache/airflow/issues/46671

Reply:
Oh I see why you were asking about base_date. We should probably get ready to remove this whole function when we delete the webserver.